easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,160 @@
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # By default, XALT automatically appends 'xalt/<version>' to the
+        # prefix, i.e., --prefix=/opt will actually install in
+        # /opt/xalt/<version>.  To precisely control the install prefix and
+        # not append anything to the prefix, use the configure option
+        # '--with-siteControlledPrefix=yes'.
+        # See https://xalt.readthedocs.io/en/latest/050_install_and_test.html
+        self.cfg.update('configopts', '--with-siteControlledPrefix=yes')
+
+        # XALT site filter config file is mandatory
+        config_py = self.cfg['config_py']
+        if config_py:
+            if os.path.exists(config_py):
+                self.cfg.update('configopts', '--with-config=%s' % config_py)
+            else:
+                raise EasyBuildError("Specified XALT configuration file %s does not exist!", config_py)
+        else:
+            error_msg = "Location of XALT configuration file must be specified via 'config_py' easyconfig parameter. "
+            error_msg += "You can edit the easyconfig file, or use 'eb --try-amend=config_py=<path>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/030_site_filtering.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # XALT system name is mandatory
+        if self.cfg['syshost']:
+            self.cfg.update('configopts', '--with-syshostConfig=%s' % self.cfg['syshost'])
+        else:
+            error_msg = "The name of the system must be specified via the 'syshost' easyconfig parameter. "
+            error_msg += "You can edit the easyconfig file, or use 'eb --try-amend=syshost=<string>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/020_site_configuration.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # Transmission method is mandatory
+        if self.cfg['transmission']:
+            self.cfg.update('configopts', '--with-transmission=%s' % self.cfg['transmission'])
+        else:
+            error_msg = "The XALT transmission method must be specified via the 'transmission' easyconfig parameter. "
+            error_msg = "You can edit the easyconfig file, or use 'eb --try-amend=transmission=<string>'. "
+            error_msg += "See https://xalt.readthedocs.io/en/latest/020_site_configuration.html for more information."
+            raise EasyBuildError(error_msg)
+
+        # GPU tracking
+        if self.cfg['gpu_tracking'] is True:
+            # User enabled
+            self.cfg.update('configopts', '--with-trackGPU=yes')
+        elif self.cfg['gpu_tracking'] is None:
+            # Default value, enable GPU tracking if nvml.h is present
+            # and the CUDA module is loaded
+            cuda_root = get_software_root('CUDA')
+            if not cuda_root:
+                # Try using the system CUDA
+                cuda_root = '/usr/local/cuda'
```

Comment:
I would not use this hardcoded fallback path, EasyBuild should not make decision based on what is installed in the OS.

Generally, CUDA is supplied via EasyBuild, and if it's not, people can still opt-in to building XALT with GPU tracking support by setting `gpu_tracking` to `True`.